src/ReadModel/JsonDocument.php

```diff
@@ -46,6 +51,11 @@ public function withBody($body)
         return new self($this->id, json_encode($body));
     }
 
+    public function withAssocBody(array $body): JsonDocument
```

Comment:
A very small nuance in naming, maybe use `fromAssocBody` to make it clear it's a named static constructor and not a mutation?

Reply:
It's not a static constructor though, it uses `$this->id`. It's basically the same as this but as a one-liner:

```
$c = clone $this;
$c->body = json_encode($body);
return $c;
```

Reply:
(And it's based on the existing `withBody()`)